kv/db.go

```diff
@@ -218,6 +69,28 @@ func (s *DBServer) RegisterRPC(rpcServer *rpc.Server) error {
 func (s *DBServer) executeCmd(argsI gogoproto.Message) (gogoproto.Message, error) {
 	args := argsI.(proto.Request)
 	reply := args.CreateReply()
+	if err := verifyRequest(args); err != nil {
+		return nil, err
+	}
 	s.sender.Send(context.TODO(), proto.Call{Args: args, Reply: reply})
 	return reply, nil
 }
+
+// verifyRequest checks for illegal inputs in request proto and
+// returns an error indicating which, if any, were found.
+func verifyRequest(args proto.Request) error {
+	switch t := args.(type) {
+	case *proto.EndTransactionRequest:
+		if t.InternalCommitTrigger != nil {
+			return util.Errorf("EndTransaction request from public KV API contains commit trigger: %+v", t.GetInternalCommitTrigger())
```

Comment:
public here is a bit of a misnomer since I locked down the KV stuff. We even has a `rpc.RegisterPublic` for endpoints that can be used by regular users.
How about changing `public` to `external` here. The various terms would now mean:
* external public: available on gateways, can be used by all users
* external private: available on gateways, can only be used by `node` user
* internal: available on nodes, can only be used by `node` user

The ones here would be "external private", although external is probably sufficient.
Keeping `verifyRequest` on them is reasonable, these are some things we do not allow to be performed through this endpoint.